src/main/java/com/oath/halodb/CompactionManager.java

```diff
@@ -44,6 +44,7 @@
     private static final int STOP_SIGNAL = -10101;
 
     private final ReentrantLock startStopLock = new ReentrantLock();
+    private final Lock compactionFileRollOverLock = new ReentrantLock();
```

Comment:
Any reason to define it as ```Lock``` instead of its actual type ```ReentrantLock```. It not required, please keep consistent to existing code. Then no need to import ```java.util.concurrent.locks.*```